pkg/ssh/connection_golang.go

```diff
@@ -70,7 +71,7 @@ func golangConnectionDial(options ConnectionDialOptions) (*ConnectionDialReport,
 	if err != nil {
 		return nil, err
 	}
-	cfg, err := ValidateAndConfigure(uri, options.Identity)
+	cfg, err := ValidateAndConfigure(uri, options.Identity, options.InsecureIsMachineConnection)
```

Comment:
LGTM. (I could go with `IsPodmanMachineConnection` with a long descriptive comment, but the `Insecure` naming warning callers is attractive.)

---

Will the other callers of `ValidateAndConfigure` need similar options? Even if they do, that can be a future PR.

Reply:
I fixed up the other callers as best as I could but only `Dial` seems to need this at the moment in podman